From 4d7d9ef2f8c2791b04328dd3e6968d7dd70322bc Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Thu, 30 Oct 2014 17:36:02 +0100
Subject: [PATCH 2/2] ethernet: Don't crash if the connection lacks wired
 setting upon activation attempt

assert() is a bit too harsh for a condition that can be triggered by an
unprivileged client. Just bail out instead.

https://bugzilla.redhat.com/show_bug.cgi?id=1156564
---
 src/nm-device-ethernet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nm-device-ethernet.c b/src/nm-device-ethernet.c
index ab2cf2f..310f0b6 100644
--- a/src/nm-device-ethernet.c
+++ b/src/nm-device-ethernet.c
@@ -1270,7 +1270,7 @@ real_act_stage1_prepare (NMDevice *dev, NMDeviceStateReason *reason)
 		g_return_val_if_fail (req != NULL, NM_ACT_STAGE_RETURN_FAILURE);
 
 		s_wired = NM_SETTING_WIRED (device_get_setting (dev, NM_TYPE_SETTING_WIRED));
-		g_assert (s_wired);
+		g_return_val_if_fail (s_wired, NM_ACT_STAGE_RETURN_FAILURE);
 
 		/* Set device MAC address if the connection wants to change it */
 		cloned_mac = nm_setting_wired_get_cloned_mac_address (s_wired);
-- 
2.1.0

